source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -100,16 +101,18 @@ bool HeaderToMetadataFilter::addMetadata(StructMap& map, const std::string& meta
   // Sane enough, add the key/value.
   switch (type) {
   case envoy::config::filter::http::header_to_metadata::v2::Config_ValueType_STRING:
-    val.set_string_value(value);
+    val.set_string_value(std::string(value));
```

Comment:
Or, we can use:

https://github.com/google/protobuf/blob/master/src/google/protobuf/struct.pb.h#L734

Would:

```
set_string_value(value.data());
```

be better?

Reply:
ignore me, it's the same thing.

Reply:
FYI value.data() would be wrong because string_view does not in general nul-terminate.  The one I'm calling will I think use the std::string&& form which should not require any more copies than the previous version, which used a const std::string&, and Value::set_string_value(const ::std::string& value) will have to copy the bytes.

I think the new version should have the same number of copies as there were previously for the string case, and fewer copies when passing to set_number_value.